### PR TITLE
Issue first graphql query from client

### PR DESCRIPTION
**Pull request by bluquar**
Screen Shot 2021-11-20 at 6 51 34 AM

### CODE REVIEW

File: App.tsx [hunk 1]
```diff
@@ -9,6 +9,14 @@ import useCachedResources from './hooks/useCachedResources';
 import useColorScheme from './hooks/useColorScheme';
 import Navigation from './navigation';
 
+import {ApolloClient, InMemoryCache, ApolloProvider} from '@apollo/client';
+
+// Initialize Apollo Client
+const client = new ApolloClient({
```
Comment (lowellbander) [inline App.tsx:15]: hehe Apollo is where that one friend of ours interviewed at recently

File: App.tsx [hunk 2]
```diff
@@ -9,6 +9,14 @@ import useCachedResources from './hooks/useCachedResources';
...
+  uri: 'http://localhost:8000/graphql',
```
Comment (lowellbander) [inline App.tsx:16]: extract to an environment variable, yeah? is this the same host as the django backend that serves up the app itself? wondering if we need two deployments for some reason

#29

File: screens/TabTwoScreen.tsx [hunk 3]
```diff
@@ -1,36 +1,67 @@
...
 import {StyleSheet} from 'react-native';
+import {ActivityIndicator, List} from 'react-native-paper';
 
 import EditScreenInfo from '../components/EditScreenInfo';
 import {Text, View} from '../components/Themed';
 
+import {gql, useQuery} from '@apollo/client';
+
+const AID_REQUESTS_QUERY = gql`
+  query {
```
Comment (lowellbander) [inline screens/TabTwoScreen.tsx:11]: so cool! u know if there's a way to enforce some formatting rules in this repo to ensure everything is formatted this prettily? #30

File: screens/TabTwoScreen.tsx [hunk 4]
```diff
@@ -1,36 +1,67 @@
...
+      createdAt
+    }
+  }
+`;
+
+type QueryResponse = {
+  allAidRequests: Array<{
+    whatIsNeeded: string;
+    completed: boolean;
+    zipCode: number;
```
Comment (lowellbander) [inline screens/TabTwoScreen.tsx:25]: this is ofc made up of numbers, but probably makes more sense as a string; we're not going to ever perform mathematical operations on it, only use it as an identifier

File: screens/TabTwoScreen.tsx [hunk 5]
```diff
@@ -1,36 +1,67 @@
...
-        darkColor="rgba(255,255,255,0.1)"
-      />
-      <EditScreenInfo path="/screens/TabTwoScreen.tsx" />
+      <div>
+        {(data as QueryResponse).allAidRequests.map(
+          ({whatIsNeeded, completed, zipCode, createdAt}, index: number) => {
+            const createdDate = new Date(createdAt);
+            return (
+              <List.Item
+                title={whatIsNeeded}
```
Comment (lowellbander) [inline screens/TabTwoScreen.tsx:47]:
```suggestion
                title={`${whatIsNeeded} (${zipCode})`}
```

**Review by lowellbander — APPROVED** [review]
outstanding! suh suh good

File: App.tsx (hunk 2 above)
Comment (bluquar) [reply App.tsx:16]: this is the host of the django backend. the app itself is currently served by a node backend (via expo), not django.

Comment (bluquar) [reply App.tsx:16]: can u teach me how environment variables work?

File: screens/TabTwoScreen.tsx (hunk 3 above)
Comment (bluquar) [reply screens/TabTwoScreen.tsx:11]: commented on the issue :)

File: screens/TabTwoScreen.tsx (hunk 4 above)
Comment (bluquar) [reply screens/TabTwoScreen.tsx:25]: sgtm!

File: screens/TabTwoScreen.tsx (hunk 5 above)
Comment (bluquar) [reply screens/TabTwoScreen.tsx:47]: #33

File: App.tsx (hunk 2 above)
Comment (lowellbander) [reply App.tsx:16]: I can try! so in my experience it's just a simple key:value file. for development, it'd be a local file (added to your .gitignore) and for deployment you configure in the UI of the deployment service.

so like for Node, you'd just have a .env file an then reference that in your JS app code: https://nodejs.dev/learn/how-to-read-environment-variables-from-nodejs

For Python, I think you might just use os.environ: https://stackoverflow.com/questions/5971312/how-to-set-environment-variables-in-python

There are also django-specific frameworks for this like https://django-environ.readthedocs.io/en/latest/

Then, for either front or backend, you can use a deployment CLI or UI to configure these, like for heroku: https://devcenter.heroku.com/articles/config-vars

File: App.tsx (hunk 1 above)
Comment (bluquar) [reply App.tsx:15]: hehe

**Pull request merged**